### ISSUE TITLE
how mdxts support remote content?

### ISSUE THREAD

**Issue by arvinxx** [souporserious/renoun#72]
I really need a method to combine with remote mdx content.

**Comment by souporserious** [comment]
I have plans to expand the `createSource` utility for this. Right now it accepts a file path for local files on disk, but if it accepts a function it could resolve to any set of files, thinking something like this:

```ts
const remoteDocs = createSource(() => fetch(...))
```

Is there a particular remote source you're looking to use or would something like above work for you?

**Comment by arvinxx** [comment]
yes, github content like https://github.com/lobehub/lobe-chat/tree/main/docs/usage

**Comment by souporserious** [comment]
Remote content is now supported through the [`MemoryFileSystem` class](https://www.renoun.dev/utilities/file-system#memory-file-system). In the future there will be first-class support for Git providers to make this easier to populate. For now you can fetch them yourself and instantiate a file system that is passed to [`Directory`](https://www.renoun.dev/utilities/file-system#directory):

```ts
import { Directory, MemoryFileSystem } from 'renoun/file-system'

const filesFromGitHubRepo = await fetchFilesFromGitHubRepo('user', 'repo') // implement yourself for now
const fileSystem = new MemoryFileSystem(filesFromGitHubRepo)
const directory = new Directory({ fileSystem })
const file = directory.getFile(...) // use file system methods
```
